app/models/photo.rb

```diff
@@ -143,4 +156,39 @@ def self.visible(current_user, person, limit=:all, max_time=nil)
              end
     photos.where(pending: false).order("created_at DESC")
   end
+
+  def coords_to_float(deg, min, sec)
+    (deg.to_r + min.to_r / 60 + sec.to_r / 3600).to_f
+  end
+
+  def determine_location(image)
+    exif_gps_latitude  = image.exif["GPSLatitude"]
+    return nil unless exif_gps_latitude
+
+    exif_gps_longitude = image.exif["GPSLongitude"]
+    return nil unless exif_gps_longitude
+
+    lat = coords_to_float(*exif_gps_latitude.split(", ")) * (image.exif["GPSLatitudeRef"] == "S" ? -1 : 1)
+    lng = coords_to_float(*exif_gps_longitude.split(", ")) * (image.exif["GPSLongitudeRef"] == "W" ? -1 : 1)
+
+    nominatim_request =
+      Typhoeus.get("https://nominatim.openstreetmap.org/reverse?" +
+                   "format=json&lat=%f&lon=%f&addressdetails=3" % [lat, lng])
+
+    address =
+      if nominatim_request.response_code == 200
+        JSON.parse(nominatim_request.response_body)["display_name"]
+      else
+        "A beautiful place."
```

Comment:
I will drop this, so that empty addresses (empty string, not null) are also allowed.